### PR TITLE
Adds tests for GroupBy with keys; removes unused Bit message from proto

### PR DESCRIPTION
**Pull request by yuce**
## Pull request checklist

- [ ] I have read the [contributing guide](https://github.com/pilosa/pilosa/blob/master/CONTRIBUTING.md).
- [ ] I have agreed to the [Contributor License Agreement](https://cla-assistant.io/pilosa/pilosa).
- [ ] I have updated the [documentation](https://github.com/pilosa/pilosa/tree/master/docs).
- [ ] I have resolved any merge conflicts.
- [ ] I have included tests that cover my changes.
- [ ] All new and existing tests pass.

## Code review checklist
This is the checklist that the reviewer will follow while reviewing your pull request. You do not need to do anything with this checklist, but be aware of what the reviewer will be looking for.

- [ ] Ensure that any changes to external docs have been included in this pull request.
- [ ] If the changes require that minor/major versions need to be updated, tag the PR appropriately.
- [ ] Ensure the new code is [properly commented](https://github.com/golang/go/wiki/CodeReviewComments#doc-comments) and follows [Idiomatic Go](https://dmitri.shuralyov.com/idiomatic-go).
- [ ] Check that tests have been written and that they cover the new functionality.
- [ ] Run tests and ensure they pass.
- [ ] Build and run the code, performing any applicable integration testing.

### CODE REVIEW

File: encoding/proto/proto.go [hunk 1]
```diff
@@ -1147,8 +1147,14 @@ func decodeGroupCounts(a []*internal.GroupCount) []pilosa.GroupCount {
...
-		other[i].Field = a[i].Field
-		other[i].RowID = a[i].RowID
+		fr := a[i]
+		other[i].Field = fr.Field
+		if fr.RowKey == "" {
+			other[i].RowID = fr.RowID
+		} else {
+			other[i].RowKey = fr.RowKey
+		}
+		fmt.Println("OTHER", other)
```
Comment (travisturner) [inline encoding/proto/proto.go]: don't need this debugging line

Comment (yuce) [reply encoding/proto/proto.go]: updated.

File: test/pilosa.go [hunk 2]
```diff
@@ -380,6 +425,17 @@ func MustDo(method, urlStr string, body string) *httpResponse {
 	return &httpResponse{Response: resp, Body: string(buf)}
 }
 
+func CheckGroupBy(t *testing.T, expected, results []pilosa.GroupCount) {
+	if len(results) != len(expected) {
+		t.Fatalf("number of  groupings mismatch:\n got:%+v\nwant:%+v\n", results, expected)
```
Comment (travisturner) [inline test/pilosa.go]: I realize this doesn't matter, and was just copied code, but there is an extra space after "of"

Comment (yuce) [reply test/pilosa.go]: updated

**Pull request merged**